code/modules/clothing/suits/armor.dm

```diff
@@ -381,8 +395,50 @@
 			if(M == owner)
 				continue
 			owner.Beam(M,icon_state="lightning[rand(1, 12)]",icon='icons/effects/effects.dmi',time=5)
-			M.adjustFireLoss(25)
+			M.adjustFireLoss(20)
 			playsound(M, 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		disable(rand(2, 5)) // let's not have buckshot set it off 4 times and do 80 burn damage.
+		return 1
+
+/obj/item/clothing/suit/armor/reactive/repulse
+	name = "reactive repulse armor"
+	desc = "An experimental suit of armor that violently throws back attackers with the power of a gravitational anomaly core."
+	///How strong the reactive armor is for throwing
+	var/repulse_power = 3
+	/// How far away are we finding things to throw
+	var/repulse_range = 5
+
+/obj/item/clothing/suit/armor/reactive/repulse/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active)
+		return 0
+	if(prob(hit_reaction_chance))
+		owner.visible_message("<span class='danger'>[src] blocks [attack_text], converting the attack into a wave of force!</span>")
```

Comment:
Doesn't make a difference here, but this could just be `visible_message` rather than `owner.visible_message`.